### PR TITLE
update taskfile and precommit configuration

### PR DESCRIPTION
**Pull request by matoszz**
- updates goreleaser to newly changed syntax for brews
- runs auto update on existing precommit hooks
- adds a default task responder that lists available tasks
- refactor of `task install` to optimize how we approach package installations to be more comprehensive

### CODE REVIEW

File: Taskfile.yaml [hunk 1]
```diff
@@ -154,3 +135,76 @@ tasks:
...
+    silent: true
+    desc: check if Homebrew is installed
+    cmds:
+      - '[ -x "$(command -v brew)" ] || (echo "Homebrew is not installed, please install it from https://brew.sh" && exit 1)'
+
+  go:dist:
+    desc: Downloads a Golang distribution
+    silent: true
+    vars:
+      GO_VERSION: 1.22.4
```
Comment (golanglemonade) [inline Taskfile.yaml:205]: Do we have to specify a patch? Can we tell it to just to get the latest minor so we don't have to constantly update this?

File: db/Taskfile.yaml [hunk 2]
```diff
@@ -22,17 +21,6 @@ tasks:
         read migration;
         go run create_migrations.go ${migration};
 
-  lint:
```
Comment (golanglemonade) [inline db/Taskfile.yaml:25]: Ca we keep this command but just not call it on `task pr`?

File: db/Taskfile.yaml [hunk 3]
```diff
@@ -12,7 +12,6 @@ tasks:
     aliases: [all]
     cmds:
       - task: create
-      - task: lint
```
Comment (golanglemonade) [inline db/Taskfile.yaml:15]: should just remove `create-and-lint` completely

File: Taskfile.yaml (hunk 1 above)
Comment (matoszz) [reply Taskfile.yaml:205]: uh... do you know of a fancy way to do that? lol

File: go.mod [hunk 4]
```diff
@@ -1,6 +1,8 @@
 module github.com/datumforge/datum
 
-go 1.22.2
+go 1.22.3
+
+toolchain go1.22.4
```
Comment (golanglemonade) [inline go.mod:5]:
```suggestion
go 1.22.4
```

I think you should just be able to do that ^ and do a `tidy`

**Pull request merged**